modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/metadata/MetadataInputDelegate.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.extension.internal.metadata;
+
+import static org.mule.runtime.api.metadata.descriptor.builder.MetadataDescriptorBuilder.parameterDescriptor;
+import static org.mule.runtime.api.metadata.resolving.MetadataResult.failure;
+import static org.mule.runtime.api.metadata.resolving.MetadataResult.mergeResults;
+import static org.mule.runtime.api.metadata.resolving.MetadataResult.success;
+import static org.mule.runtime.module.extension.internal.util.MetadataTypeUtils.subTypesUnion;
+import org.mule.metadata.api.model.MetadataType;
+import org.mule.runtime.api.metadata.MetadataContext;
+import org.mule.runtime.api.metadata.MetadataKey;
+import org.mule.runtime.api.metadata.descriptor.InputMetadataDescriptor;
+import org.mule.runtime.api.metadata.descriptor.ParameterMetadataDescriptor;
+import org.mule.runtime.api.metadata.descriptor.TypeMetadataDescriptor;
+import org.mule.runtime.api.metadata.descriptor.builder.InputMetadataDescriptorBuilder;
+import org.mule.runtime.api.metadata.descriptor.builder.MetadataDescriptorBuilder;
+import org.mule.runtime.api.metadata.resolving.InputTypeResolver;
+import org.mule.runtime.api.metadata.resolving.MetadataResult;
+import org.mule.runtime.extension.api.annotation.metadata.Content;
+import org.mule.runtime.extension.api.introspection.RuntimeComponentModel;
+import org.mule.runtime.extension.api.introspection.parameter.ParameterModel;
+import org.mule.runtime.extension.api.util.SubTypesMappingContainer;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Metadata service delegate implementations that handles the resolution
+ * of a {@link RuntimeComponentModel} {@link InputMetadataDescriptor}
+ *
+ * @since 4.0
+ */
+class MetadataInputDelegate extends BaseMetadataDelegate {
+
+
+  private final SubTypesMappingContainer typesMapping;
+
+  MetadataInputDelegate(RuntimeComponentModel componentModel, SubTypesMappingContainer subTypesMappingContainer) {
+    super(componentModel);
+    typesMapping = subTypesMappingContainer;
+  }
+
+  /**
+   * For each of the Component's {@link ParameterModel} creates the corresponding {@link TypeMetadataDescriptor} using only its
+   * static {@link MetadataType} and ignoring if any parameter has a dynamic type.
+   *
+   * @return A {@link List} containing a {@link MetadataResult} of {@link TypeMetadataDescriptor} for each input parameter using
+   * only its static {@link MetadataType} and ignoring if any parameter has a dynamic type.
+   */
+  MetadataResult<InputMetadataDescriptor> getInputMetadataDescriptors(MetadataContext context, Object key) {
+    InputMetadataDescriptorBuilder input = MetadataDescriptorBuilder.inputDescriptor();
```

Comment:
inputDescriptor() is not a good name for this... it should be newBuilder() or something like that